### ISSUE TITLE
Some channel messages never load, even mentions

### ISSUE THREAD

**Issue by 86doteth** [status-im/status-react#21581]
# Bug Report

## Problem 
[comment]: # (A problem description)
this is probably tied to slow message loading in general. but the problem is some messages for which i can see on desktop that they arrived in certain channels of a community (for example status community) dont arrive in mobile even days later. of course i havent kept my the app open all this time, maybe for a total of 30 mins over those days? one of those messages was even a mention by felicio of me in the web channel and i did get a push notification for it and even a badge in the app on the status community but when opening the community no channel was highlighted and when ive checked the web channel at different times since (and tapping ‘fetch messages’ many times) the channel doesnt show the message ( and many others)

#### Expected behavior
channel shows all messages and at the very least messages in which im tagged

#### Actual behavior
channel shows few messages and sometimes not even mentions of me

### Reproduction

1. dont open the app for a day or so so there are a good few dozens of missed messages in different communities like status and status inu
2. open the app and wait for a few minutes and compare with desktop how many messages are missing. tap ‘fetch messages’ a few times
3. open the app a few more times in the following days and try ‘fetch messages’ a few more times as well and notice slight improvement but still lots of community messages missing

### Additional Information

- Status version: 2.30 
- Operating System: iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by churik** [comment]
I can confirm the issue for status community, check out the video for `memes` channel, though I'm not sure how to reproduce it.

https://github.com/user-attachments/assets/02a2ee18-8cc2-4e5c-96a9-69f90b5ad6dc


Also I noticed that the unread dot is shown before the message actually arrives, so in case of Status community it can take 2-3 mins when you open the channel and see the actual message.

I was unable to reproduce it (when message completely did not show up) with new users, seems the issue is affected users who are already joined community and did not open it for a couple of days, like it was in description

Thanks for reporting @86doteth 
cc @ilmotta

**Comment by ilmotta** [comment]
To me sometimes it's the opposite as described in this issue. For example, right now I see the last message in Discord and in the mobile app match, but on Desktop it's showing older messages, even after using the `Fetch messages` option from the channel context menu. I'm using Desktop 2.30.0, so not the latest, similar to this issue (I should definitely upgrade my Desktop to latest). Actually no matter what I do, my latest messages in the `feedback-mobile` channel are out of sync in the mobile and desktop clients (some are missing on Mobile, some are missing on Desktop). This is a recurring problem.

-----

About the `Fetch messages` feature, I think it should not be available to end users. It was supposed to be a debugging utility for developers as far as I remember (see commit https://github.com/status-im/status-mobile/pull/18115/commits/32cfd214ca6091087e7ffc1f99085e04caa2182e). Many times it doesn't seem to work for me, at least not in a timely manner (takes ages and I just move on to something else). This "feature" is creating the incorrect expectations to users (cc @xAlisher). I remember we used to say fetching messages manually makes us developers "lazy" because we tend to just fetch manually and not look into the actual root cause.

@jrainville how's the experience on Desktop for the feature `Fetch messages`? Maybe it should be available as a toggle in advanced settings and disabled by default. On Mobile it's already a toggle, but it's fixed at build time and defaults to enabled.

**Comment by 86doteth** [comment]
> To me sometimes it's the opposite as described in this issue.

actually same for me. the ‘gold standard’ for me seems to be my always-on ubuntu desktop instance that has relay turned on. that one seems to have most consistently the most complete message set. 

my status desktop on macbook which is set to light node and my mobile app seem to be both roughly as bad in amount of missing messages, where a restart for the macbook app seems to help to get a more complete message set. but yeah sometimes there are streaks where mobile seems to be getting all recent messages and reacts and macbook is missing a lot and other times vice versa.

everything on latest release here

**Comment by jrainville** [comment]
> @jrainville how's the experience on Desktop for the feature Fetch messages? Maybe it should be available as a toggle in advanced settings and disabled by default. On Mobile it's already a toggle, but it's fixed at build time and defaults to enabled.

I think it is a Debug only option on Desktop 

![Image](https://github.com/user-attachments/assets/beaee807-b60d-4626-8b63-c04c78fd9fc6)

Indeed, I don,t think it was intended as a user facing feature and it doesn't even work very well, so I don't think we should rely on it.

**Comment by ilmotta** [comment]
Thanks @jrainville. I think desktop is a bit better regarding this "feature" because in mobile it's not a debug only option.

**Comment by churik** [comment]
but also there are "Fetch messages" in a chat history on desktop, which also does not work for me (it was working on mobile on v1, on v2 it just adds noise)
We do not have it on mobile now. 

cc @jrainville @ilmotta 

https://github.com/user-attachments/assets/87c4e8ef-f3ec-4e6d-8acc-17e1a862ebf6

**Comment by jrainville** [comment]
> but also there are "Fetch messages" in a chat history on desktop, which also does not work for me (it was working on mobile on v1, on v2 it just adds noise)

Right, this is one of the things almost no one understands. I know they are called "gaps". I don't remember why they happen and what clicking on it does.

Maybe we should just remove it indeed, because it's impossible to convey to the users correctly what it even means.
In theory, having good automatic background fetching should fix missing messages in the middle and it shouldn't require user actions.

**Comment by churik** [comment]
I know the meaning and they were working on v1 protocols for 1-1 and group chats; but never for v2 unfortunately

Thanks for comment @jrainville

**Comment by churik** [comment]
It should be significantly improved in 2.33 for mobile clients

**Comment by ilmotta** [comment]
I second the seriousness of the bad experience reported in this issue. The issue was reported for Status 2.30.0, but I experienced at least one missing message in the Status community in 2.32.0 as well in recent times. Could it be related to issues in the Status control node that seem to have plagued us since late last year (?)

Although there are no steps to deterministically reproduce the bug, for one of the CCs to tackle this bug, it may be useful to dogfood for a while in a dev build until they face the same bug, because then they will be able to debug when the bug flies on screen. Given how frequently we spot bugs in community messages, I believe this approach could work/help.

@churik We need to prioritize together the outstanding bugs in communities for the upcoming 2.34 lifecycle. There's a chance the fix/improvements could impact desktop as well because I suppose the root cause of this bug is in status-go (not in client code).

**Comment by ilmotta** [comment]
- This issue's root cause might be related to https://github.com/status-im/status-mobile/issues/21906
